File: code-examples/pymongo/indexes/single-field.py
# Import MongoClient from pymongo
from pymongo import MongoClient

# Replace the following with your MongoDB deployment's connection string
uri = "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority"

client = MongoClient(uri)

# Connect to your 'sample_mflix' database and 'movies' collection
database = client['sample_mflix']
movies = database['movies']

# Create an ascending index on the "title" field in the "movies" collection
result = movies.create_index('title')
print(f'Index created: {result}')

# Define the query parameters
query = { 'title': 'Batman' }
sort = [('title', 1)]
projection = { '_id': 0, 'title': 1 }

# Execute the query using the defined parameters
cursor = movies.find(query, projection).sort(sort)

# Close the connection to MongoDB
client.close()

File: code-examples/java-rs/indexes/single-field.java
// Import required libraries
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoCollection;
import com.mongodb.reactivestreams.client.MongoDatabase;
import org.bson.Document;
import com.mongodb.client.model.Indexes;
import com.mongodb.client.model.Projections;
import com.mongodb.client.model.Sorts;

import java.util.Optional;

public class Main {
    // Replace the following with your MongoDB deployment's connection string
    private static final String connectionString = "mongodb+srv://<user>:<password>@<cluster-url>?writeConcern=majority";
    
    public static void main(String[] args) {
        // creating single field indexes
        try (MongoClient mongoClient = MongoClients.create(connectionString)){
            MongoDatabase sample_mflix = mongoClient.getDatabase("sample_mflix");
            MongoCollection<Document> movies = sample_mflix.getCollection("movies");

            // Create an ascending index on the "title" field in the "movies" collection
            movies.createIndex(Indexes.ascending("title")).subscribe(indexName -> System.out.println("Index created: " + indexName));
            
            // Define the query parameters
            Document query = new Document("title", "Batman");
            movies.find(query)
                    .sort(Sorts.ascending("title"))
                    .projection(Projections.include("title"))
                    .subscribe(
                            result -> System.out.println(result.toJson()),
                            Throwable::printStackTrace,
                            () -> System.out.println("Completed")
                    );
            Thread.sleep(5000);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
